### ISSUE TITLE
Unknown symbol 'fileno' on Windows

### ISSUE THREAD

**Issue by mrkkrp** [snoyberg/bzlib-conduit#3]
I ran `stack test` on a project that has `bzlib-conduit` as a transitive dependency (on Windows), and here is what I got:

```
ghc.EXE: unable to load package `bzlib-conduit-0.2.1.4'
ghc.EXE: C:\sr\snapshots\2af6cf35\lib\x86_64-windows-ghc-8.0.1\bzlib-conduit-0.2.1.4-FcuRUOYtGcc9ouynQ0N6Js\HSbzlib-conduit-0.2.1.4-FcuRUOYtGcc9ouynQ0N6Js.o: unknown symbol `fileno'
```

Any idea what's wrong?

**Comment by snoyberg** [comment]
No, I haven't touched the C code at all. Does normal usage of the library
work on Windows?

**Comment by mrkkrp** [comment]
I didn't try to run it except for tests (a user reported issues that are only present on Windows, eh), but my guess is that this will happen every time GHC needs to load the package to do actual work. I'll return to this later. If you wish to request “experimental data”, I can do the experiments :-)

**Comment by mrkkrp** [comment]
For some people `stack test` on that library certainly works though.

**Comment by snoyberg** [comment]
I don't really know where to begin on debugging this, as I didn't write this library, just took over maintainership. If you have any thoughts on where to start, please go for it :)

**Comment by mrkkrp** [comment]
Here is a blog post (in Japanese though) that seems to be relevant:

https://kurokawh.blogspot.ru/2015/03/windowshaskell-bzlibhsbzlib-0505o.html

According to it, the library uses `fileno` and `fdopen` from `mingw` (specifically `libmoldname.a`), but in my case they are not linked. I'll try to experiment with this.

The author proposes a patch:

https://raw.githubusercontent.com/kurokawh/work_haskell/master/etc/codec/bzip2/bzlib.patch
